akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedSettings.scala

```diff
@@ -49,12 +51,25 @@ import com.typesafe.config.Config
       snapshotPluginId)
   }
 
-  private[akka] final def journalConfigFor(config: Config, journalPluginId: String): Config = {
-    val defaultJournalPluginId = config.getString("akka.persistence.journal.plugin")
+  private def journalConfigFor(config: Config, journalPluginId: String): Config = {
+    def defaultJournalPluginId = config.getString("akka.persistence.journal.plugin") match {
+      case "" =>
+        throw new IllegalArgumentException(
+          s"Default journal plugin is not configured, " +
+          "see 'akka.persistence.journal.plugin' in reference.conf.")
+      case pluginId => pluginId
+    }
     val configPath = if (journalPluginId == "") defaultJournalPluginId else journalPluginId
+    if (!config.hasPath(configPath))
+      throw new IllegalArgumentException(s"Journal plugin [$configPath] is not configured.")
     config.getConfig(configPath).withFallback(config.getConfig(Persistence.JournalFallbackConfigPath))
   }
 
+  private def verifySnapshotConfig(config: Config, snapshotPluginId: String): Unit = {
+    if (snapshotPluginId != "" && !config.hasPath(snapshotPluginId))
+      throw new IllegalArgumentException(s"Snapshot store plugin [$snapshotPluginId] is not configured.")
+  }
+
```

Comment:
Nit that these checks are sprinkled throughout the codebase. WDYT about https://github.com/akka/akka/pull/26559

Reply:
@helena ok, I extracted some common validation: 5dc23e3